### PR TITLE
Kyle Schulz

### PR DESCRIPTION
**Pull request by kylejschulz**

### CODE REVIEW

File: lib/character.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Character
+  attr_reader :name, :actor, :salary
+  def initialize(args)
+    @name = args[:name]
+    @actor = args[:actor]
+    @salary = args[:salary]
+  end
+
+  def main_character
```
Comment (robertguturing) [inline lib/character.rb:9]: This is a really great helper method and I'm happy to see it.  One minor nitpick is if you have a method that only returns boolean values then by ruby convention the method name should have a `?` in its name

File: lib/show.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  def actors
+    all_actors = []
+    @characters.each do |character|
+      all_actors << character.actor
+    end
+    all_actors
+  end
+
+  def show_main_characters
+    main_chars = @characters.select do |char|
```
Comment (robertguturing) [inline lib/show.rb:33]: Good to see you following SRP here by using the helper method you made above 🙌

File: lib/show.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  attr_reader :name, :creator, :characters
+  def initialize(name, creator, characters)
+    @name = name
+    @creator = creator
+    @characters = characters
+  end
+
+  def total_salary
+    sum = 0
+    @characters.each do |character|
```
Comment (robertguturing) [inline lib/show.rb:11]: You may not have learned this yet but I would use `reduce` here since we are returning a single value

**Review by robertguturing — COMMENTED** [review]
Nice use of enumerables and following SRP

Score : 3

**Comment by kylejschulz** [comment]
Thanks Bob. I def need to work through the enumerables ruby exercise and lean into more powerful/specific ones.

**Pull request closed without merging**